app/controllers/api/v1/mixins/rbac_mixin.rb

```diff
@@ -121,38 +125,31 @@ def owner_request_ids
         end
 
         # All child request ids for approver to process
-        def all_request_ids_for_approver
-          Request.where(:workflow_id => workflow_ids).pluck(:id).sort
-        end
-
         def visible_request_ids_for_approver
-          request_ids = all_request_ids_for_approver
-          Rails.logger.debug { "Approvable request ids: #{request_ids}" }
-
-          group_refs = assigned_group_refs
-          Rails.logger.debug { "Groups from assigned roles: #{group_refs}" }
-
           visible_states = [ApprovalStates::NOTIFIED_STATE, ApprovalStates::COMPLETED_STATE]
-          Request.where(:id => request_ids, :group_ref => group_refs, :state => visible_states).pluck(:id)
+          Request.where(:workflow_id => workflow_ids, :state => visible_states).pluck(:id)
         end
 
         def assigned_group_refs
-          assigned_roles.map { |name, _id| name.split(AccessProcessService::APPROVER_ROLE_PREFIX)[1] }.compact
+          Insights::API::Common::RBAC::Service.call(RBACApiClient::GroupApi) do |api|
+            Insights::API::Common::RBAC::Service.paginate(api, :list_groups, :scope => 'principal').collect(&:uuid)
+          end
         end
 
         # The accessible workflow ids for approver
         def workflow_ids
-          approval_access = Insights::API::Common::RBAC::Access.new('workflows', 'approve').process
-
-          Rails.logger.debug { "Approvable workflows: #{approval_access.id_list}" }
-
-          approval_access.id_list
+          AccessControlEntry.where(:aceable_type => 'Workflow', :permission => 'approve', :group_uuid => assigned_group_refs).pluck(:aceable_id)
         end
 
         # The access list regular requesters have
         def requester_acls
           Insights::API::Common::RBAC::ACL.new.create(nil, OWNER_PERMISSIONS)
         end
+
+        # Access list approvers can have
+        def approver_acls
+          Insights::API::Common::RBAC::ACL.new.create(nil, APPROVER_PERMISSIONS)
```

Comment:
@bzwei Do we need to be using this class, in catalog we are getting rid of this class from the common gem.
Since the ACE's are stored locally. You only need permissions from the RBAC service now

Reply:
https://github.com/RedHatInsights/insights-api-common-rails/blob/master/lib/insights/api/common/rbac/access.rb

Reply:
We are trying to get rid of this one
https://github.com/RedHatInsights/insights-api-common-rails/blob/master/lib/insights/api/common/rbac/acl.rb

Which did all of the heavy lifting for managing ACL's on the RBAC server side

Reply:
We use this
https://github.com/RedHatInsights/catalog-api/blob/cebf916601c3df35e15c5fbb53a748458e86578e/app/controllers/api/v1/mixins/rbac_mixin.rb#L37
Which does the resolution on the RBAC service and gets us the base Access Object.

Reply:
Or
https://github.com/RedHatInsights/catalog-api/blob/cebf916601c3df35e15c5fbb53a748458e86578e/app/controllers/api/v1/mixins/index_mixin.rb#L33